Dockerfile

```diff
@@ -36,25 +36,30 @@ RUN mkdir -p ${HADOOP_HOME} && mkdir -p ${SPARK_HOME}
 WORKDIR ${SPARK_HOME}
 
 # Download spark
+# see resources: https://dlcdn.apache.org/spark/spark-3.5.5/
+# filename: spark-3.5.5-bin-hadoop3.tgz 
 RUN mkdir -p ${SPARK_HOME} \
     && curl https://dlcdn.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz -o spark-${SPARK_VERSION}-bin-hadoop3.tgz \
-    && tar xvzf spark-${SPARK_VERSION}-bin-hadoop3.tgz --directory /opt/spark --strip-components 1 \
+    && tar xvzf spark-${SPARK_VERSION}-bin-hadoop3.tgz --directory ${SPARK_HOME} --strip-components 1 \
     && rm -rf spark-${SPARK_VERSION}-bin-hadoop3.tgz
 
+# Add spark binaries to shell and enable execution
+RUN chmod u+x /opt/spark/sbin/* && \
+    chmod u+x /opt/spark/bin/*
+ENV PATH="$PATH:$SPARK_HOME/bin:$SPARK_HOME/sbin"
+
+# Add a spark config for all nodes
+COPY conf/spark-defaults.conf "$SPARK_HOME/conf/"
+
 
-FROM spark-base as pyspark
+FROM spark-base AS pyspark
 
 # Install python deps
 COPY requirements.txt .
 RUN pip3 install -r requirements.txt
 
-COPY conf/spark-defaults.conf "$SPARK_HOME/conf"
```

Comment:
an absense of backslash on the end here results a `conf` file creation instead of `spark-defaults.conf`